src/main/java/com/solace/spring/cloud/stream/binder/inbound/queue/JCSMPInboundQueueMessageProducer.java

```diff
@@ -121,7 +121,11 @@ private void sendToConsumer(Message<?> message) {
         if (deliveryAttempt != null) {
             deliveryAttempt.incrementAndGet();
         }
+        long beforeMessageProcessing = System.nanoTime();
         sendMessage(message);
+        long afterMessageProcessing = System.nanoTime();
+        solaceMeterAccessor.ifPresent(meterAccessor -> meterAccessor.recordMessageProcessingTimeDuration(consumerProperties.getBindingName(),
```

Comment:
nano to milli sec is: 1000000 not 1000

Reply:
Use TimeUnit.NANO.toMillis(...) if you want to truncate it ( like (int) ) and enforce floating point operation if you want it precise (afterMessageProcessing - beforeMessageProcessing) / (1000.0 * 1000)